site/app/templates/submission/homework/BulkUploadBox.twig

```diff
@@ -112,24 +112,75 @@
         $("#messages").empty();
     });
 
+    //gets the raw value entered in the expected page box
+    function getExpectedPageBoxVal(){
+        var expected_count_box = document.getElementById("expected-pages-input");
+        if (!expected_count_box)
+            return NaN;
+
+        return expected_count_box.value;
+    }
+
+    //parses the expected page box and returns a list of expected pages
+    //accepts multiple pages and also ranges, e.g 1-5,8,8-13
+    function parseExpectedPageBox(){
+        var expected_count = getExpectedPageBoxVal();
+        if ( isNaN(expected_count) || expected_count === "" || expected_count === null)
+            return NaN;
+
+        parts = expected_count.split(',');
+        ret = [];
+
+        for (var i = 0; i < parts.length; i++) {
+            if(parts[i].length > 1 && parts[i].includes('-')){
+                //handle range
+                tmp = parts[i].split('-');
+                if (tmp.length != 2)  
+                    continue;
+
+                start = parseInt(tmp[0]);
+                end = parseInt(tmp[1]);
+
+                if (isNaN(start) || isNaN(end) || start < 0 || end < start)
+                    continue;
+
+                for(var j = start; j <= end; j++)
+                    ret.push(j);
+
+            }else{
+                val = parseInt(parts[i]);
+                if (isNaN(val) || val < 0)
+                    continue;
+
+                ret.push( val );
+            }
+        }
+
+        return ret;
+    }
+
     //highlight pages that don't match the expected, also save the expected page count in session storage
     function highlightPageCount(){
-        var expected_count = parseInt(document.getElementById("expected_pages_input").value);
-        if(expected_count == NaN || !expected_count)
+
+        expected_count = parseExpectedPageBox();
+
+        if (expected_count === [] || isNaN(expected_count) )
             return;
 
         var pages = document.getElementsByClassName("pdf_page_count");
         for (var i = pages.length - 1; i >= 0; i--) {
-            if(parseInt(pages[i].innerHTML) !== expected_count && expected_count > 0)
+            if(expected_count.indexOf( parseInt(pages[i].innerHTML) ) === -1 )
                 pages[i].style.backgroundColor = "#FF5722";
             else
                 pages[i].style.backgroundColor = "";
         }
 
+        //use the raw value entered by the user to prevent a conversion, e.g
+        //we don't want to store 1-3 as 1,2,3
         if(!sessionStorage.getItem("expected_count")){
-            sessionStorage.setItem("expected_count", expected_count);
+            sessionStorage.setItem("expected_count", getExpectedPageBoxVal());
         }
-        sessionStorage.setItem("expected_count", expected_count);
+        sessionStorage.setItem("expected_count", getExpectedPageBoxVal());
```

Comment:
I think we can remove this if check and just do a singular `sessionStorage.setItem("expected_count", getExpectedPageBoxVal());`.